File: engine/base/src/main/java/tools/packet/buff/GiveForeignSlowDebuff.java
package tools.packet.buff;

import java.util.List;

import client.MapleDisease;
import net.opcodes.SendOpcode;
import server.life.MobSkill;
import tools.Pair;
import tools.packet.PacketInput;

public class GiveForeignSlowDebuff implements PacketInput {
   private int characterId;

   private List<Pair<MapleDisease, Integer>> statups;

   private MobSkill mobSkill;

   public GiveForeignSlowDebuff(int characterId, List<Pair<MapleDisease, Integer>> statups, MobSkill mobSkill) {
      this.characterId = characterId;
      this.statups = statups;
      this.mobSkill = mobSkill;
   }

   @Override
   public SendOpcode opcode() {
      return SendOpcode.GIVE_FOREIGN_BUFF;
   }

   public int getCharacterId() {
      return characterId;
   }

   public List<Pair<MapleDisease, Integer>> getStatups() {
      return statups;
   }

   public MobSkill getMobSkill() {
      return mobSkill;
   }
}

File: engine/engine-model/src/main/scala/tools/packet/buff/GiveForeignSlowDebuff.scala
package tools.packet.buff

import client.MapleDisease
import net.opcodes.SendOpcode
import server.life.MobSkill
import tools.Pair
import tools.packet.PacketInput

class GiveForeignSlowDebuff(private var _characterId: Int,
                            private var _statups: java.util.List[Pair[MapleDisease, java.lang.Integer]],
                            private var _mobSkill: MobSkill) extends PacketInput {
  def characterId: Int = _characterId

  def statups: java.util.List[Pair[MapleDisease, java.lang.Integer]] = _statups

  def mobSkill: MobSkill = _mobSkill

  override def opcode(): SendOpcode = SendOpcode.GIVE_FOREIGN_BUFF
}